pkg/issuer/acme/issue.go

```diff
@@ -218,13 +217,7 @@ func (a *Acme) Issue(ctx context.Context, crt *v1alpha1.Certificate) (*issuer.Is
 func (a *Acme) cleanupOwnedOrders(ctx context.Context, crt *v1alpha1.Certificate, retain string) error {
 	log := logf.FromContext(ctx)
 
-	// TODO: don't use a label selector at all here, instead we can index orders by their ownerRef and query based on owner reference alone
-	// construct a label selector
-	req, err := labels.NewRequirement(certificateNameLabelKey, selection.Equals, []string{crt.Name})
-	if err != nil {
-		return err
-	}
-	selector := labels.NewSelector().Add(*req)
+	selector := labels.NewSelector()
 
 	existingOrders, err := a.orderLister.Orders(crt.Namespace).List(selector)
```

Comment:
I think we can just use `selector.Everything()` here instead 😄

Reply:
I tried that but I believe the Lister gets a labels.Selector, which doesn't provide Everything (https://github.com/jetstack/cert-manager/blob/master/pkg/client/listers/certmanager/v1alpha1/order.go#L48) - I may very well be misreading though!